--- a/src/fbdev-orig.c	2019-09-03 16:06:26.281167077 +0200
+++ b/src/fbdev.c	2019-09-03 16:08:56.893215757 +0200
@@ -125,7 +125,8 @@
 	OPTION_SHADOW_FB,
 	OPTION_ROTATE,
 	OPTION_FBDEV,
-	OPTION_DEBUG
+	OPTION_DEBUG,
+	OPTION_DIMM
 } FBDevOpts;
 
 static const OptionInfoRec FBDevOptions[] = {
@@ -133,6 +134,7 @@
 	{ OPTION_ROTATE,	"Rotate",	OPTV_STRING,	{0},	FALSE },
 	{ OPTION_FBDEV,		"fbdev",	OPTV_STRING,	{0},	FALSE },
 	{ OPTION_DEBUG,		"debug",	OPTV_BOOLEAN,	{0},	FALSE },
+	{ OPTION_DIMM,		"dimm", 	OPTV_BOOLEAN,	{0},	FALSE },
 	{ -1,			NULL,		OPTV_NONE,	{0},	FALSE }
 };
 
@@ -1004,7 +1006,11 @@
 
 	xf86DPMSInit(pScreen, fbdevDPMSSet, 0);
 
-	pScreen->SaveScreen = fbdevSaveScreen;
+	if (xf86ReturnOptValBool(fPtr->Options, OPTION_DIMM, FALSE)) {
+		xf86DrvMsg(pScrn->scrnIndex, X_CONFIG, "backlight dimm option active\n");
+		pScreen->SaveScreen = FBSaveScreen;
+	} else
+		pScreen->SaveScreen = fbdevHWSaveScreenWeak();
 
 	/* Wrap the current CloseScreen function */
 	fPtr->CloseScreen = pScreen->CloseScreen;
@@ -1026,6 +1032,18 @@
 	return TRUE;
 }
 
+static Bool
+FBSaveScreen(ScreenPtr pScreen, int mode)
+{
+	Bool on = xf86IsUnblank(mode);
+
+	if (mode == 0)
+		system("echo 1 > /sys/class/backlight/backlight/brightness");
+	if (mode == 1)
+		system("echo 16 > /sys/class/backlight/backlight/brightness");
+	return TRUE;
+}
+
 static Bool
 FBDevCloseScreen(CLOSE_SCREEN_ARGS_DECL)
 {
