--- pil21/src/Makefile	2025-09-06 16:28:40.000000000 +0200
+++ pil21/src/Makefile	2025-09-10 15:29:41.168078932 +0200
@@ -1,22 +1,17 @@
 # 06sep25 Software Lab. Alexander Burger
 
-.SILENT:
 
-CC = clang
-PIL = ../pil  # pil
-ASM = opt -O2  # llvm-as
-LLC = llc
-LINK = llvm-link
+CC ?= clang
+PIL ?= ../pil  # pil
+ASM ?= opt -O2  # llvm-as
+LLC ?= llc
+LINK ?= llvm-link
 SHARED = -shared
-STRIP = strip
+STRIP ?= strip
+LLVM_CONFIG ?= llvm-config
 
-LLVM = $(shell llvm-config --version | cut -d. -f1)
-ifeq ($(shell test $(LLVM) -ge 15 -a $(LLVM) -lt 17; echo $$?), 0)
-	ASM += -opaque-pointers
-endif
-
-OS = $(shell uname)
-CPU = $(shell uname -m)
+OS ?= $(shell uname)
+CPU ?= $(shell uname -m)
 
 BIN = ../bin
 LIB = ../lib
@@ -29,7 +24,7 @@
 
 # System definitions
 $(LIB)/sysdefs: sysdefs.c
-	$(CC) -w -D_OS='"$(OS)"' -D_CPU='"$(CPU)"' sysdefs.c  &&  ./a.out > $(LIB)/sysdefs  &&  rm ./a.out
+	$(CC) -w -D_OS='"$(OS)"' -D_CPU='"$(CPU)"' sysdefs.c -o sysdef
 
 # Base system
 $(BIN)/picolisp: picolisp.bc
@@ -54,7 +49,7 @@
 # Base system as shared library
 $(LIB)/picolisp.so: picolisp.so.bc
 	$(LLC) picolisp.so.bc -relocation-model=pic -o picolisp.so.s
-	$(CC) picolisp.so.s -o $(LIB)/picolisp.so $(SHARED)
+	$(CC) picolisp.so.s -o $(LIB)/picolisp.so $(SHARED) -Wl,-soname,libpicolisp.so.1
 	$(STRIP) $(LIB)/picolisp.so
 
 picolisp.so.bc: base.so.bc lib.so.bc
