--- a/src/omapfb-driver.c	2015-10-01 09:01:04.933064864 +0200
+++ b/src/omapfb-driver.c-new	2015-10-01 09:09:27.850531133 +0200
@@ -76,6 +76,7 @@
 static Bool OMAPFBEnterVT(VT_FUNC_ARGS_DECL);
 static void OMAPFBLeaveVT(VT_FUNC_ARGS_DECL);
 static Bool OMAPFBSwitchMode(SWITCH_MODE_ARGS_DECL);
+static Bool OMAPFBSaveScreen(ScreenPtr pScreen, int mode);
 
 static Bool
 OMAPFBEnsureRec(ScrnInfoPtr pScrn)
@@ -644,6 +645,8 @@
 	/* Initialize RANDR support */
 	xf86CrtcScreenInit(pScreen);
 
+	pScreen->SaveScreen = OMAPFBSaveScreen;
+
 	return TRUE;
 }
 
@@ -715,6 +718,18 @@
 }
 
 static Bool
+OMAPFBSaveScreen(ScreenPtr pScreen, int mode)
+{
+	Bool on = xf86IsUnblank(mode);
+
+	if (mode == 0)
+		system("echo 1 > /sys/class/backlight/backlight/brightness");
+	if (mode == 1)
+		system("echo 16 > /sys/class/backlight/backlight/brightness");
+	return TRUE;
+}
+
+static Bool
 OMAPFBDriverFunc(ScrnInfoPtr pScrn, xorgDriverFuncOp op, pointer ptr)
 {
 	xorgHWFlags *flag;
