From 38f67441a120f91ab5881c61f95b0250e5a20139 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robert=20W=C3=B6rle?= <rwoerle@mibtec.de>
Date: Mon, 7 Oct 2024 11:24:50 +0000
Subject: [PATCH 2/2] fix to new get_user_pages()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Robert Wörle <rwoerle@mibtec.de>
---
 services4/srvkm/env/linux/osfunc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services4/srvkm/env/linux/osfunc.c b/services4/srvkm/env/linux/osfunc.c
index de0e7ae..7cc2c91 100644
--- a/services4/srvkm/env/linux/osfunc.c
+++ b/services4/srvkm/env/linux/osfunc.c
@@ -3622,7 +3622,7 @@ PVRSRV_ERROR OSAcquirePhysPageAddr(IMG_VOID *pvCPUVAddr,
 
     /* Get page list */
     psInfo->iNumPagesMapped = get_user_pages(
-		uStartAddr, psInfo->iNumPages, 1, psInfo->ppsPages, NULL);
+		uStartAddr, psInfo->iNumPages, 1, psInfo->ppsPages);
 
     if (psInfo->iNumPagesMapped >= 0)
     {
-- 
2.34.1

