From 823ee04a023aceab0445aa1783b56c43914dd8bd Mon Sep 17 00:00:00 2001
From: Robert Woerle <robert@linuxdevelopment.de>
Date: Wed, 3 May 2017 16:20:44 +0200
Subject: [PATCH] test-patch

Signed-off-by: Robert Woerle <robert@linuxdevelopment.de>
---
 src/compositor-drm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index c3f5f7f..8957bd5 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -531,6 +531,8 @@ drm_output_prepare_scanout_view(struct drm_output *output,
 	if (ev->geometry.scissor_enabled)
 		return NULL;
 
+	return NULL;
+
 	bo = gbm_bo_import(b->gbm, GBM_BO_IMPORT_WL_BUFFER,
 			   buffer->resource, GBM_BO_USE_SCANOUT);
 
-- 
1.9.1

