From 9099742e461c7affe8a6e78c0f1ec186dc8baec0 Mon Sep 17 00:00:00 2001
From: Robert Woerle <robert@linuxdevelopment.de>
Date: Wed, 3 May 2017 13:45:11 +0200
Subject: [PATCH] do not use IMPORT_WL_BUFFER but IMPORT_FD

Signed-off-by: Robert Woerle <robert@linuxdevelopment.de>
---
 src/compositor-drm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index c3f5f7f..c521f11 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -531,7 +532,7 @@ drm_output_prepare_scanout_view(struct drm_output *output,
 	if (ev->geometry.scissor_enabled)
 		return NULL;
 
-	bo = gbm_bo_import(b->gbm, GBM_BO_IMPORT_WL_BUFFER,
-                          buffer->resource, GBM_BO_USE_SCANOUT);
+/*	bo = gbm_bo_import(b->gbm, GBM_BO_IMPORT_FD,
+ 			   buffer->resource, GBM_BO_USE_SCANOUT);*/
 
 	/* Unable to use the buffer for scanout */
@@ -1034,7 +1034,7 @@ drm_output_prepare_overlay_view(struct drm_output *output,
 		return NULL;
 #endif
 	} else {
-		bo = gbm_bo_import(b->gbm, GBM_BO_IMPORT_WL_BUFFER,
+		bo = gbm_bo_import(b->gbm, GBM_BO_IMPORT_FD,
 				   buffer_resource, GBM_BO_USE_SCANOUT);
 	}
 	if (!bo)
-- 
1.9.1

