From f0eefa7a6a9c5443242aefce870fa7fba1103125 Mon Sep 17 00:00:00 2001
From: Robert Woerle <robert@linuxdevelopment.de>
Date: Wed, 15 Nov 2017 08:53:18 +0100
Subject: [PATCH] disable-security-check-on-screenshooter

Signed-off-by: Robert Woerle <robert@linuxdevelopment.de>
---
 screenshooter.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/screenshooter.c b/screenshooter.c
index d54b47d..b40171e 100644
--- a/screenshooter.c
+++ b/screenshooter.c
@@ -266,11 +266,11 @@ bind_shooter(struct wl_client *client,
 	resource = wl_resource_create(client,
 				      &screenshooter_interface, 1, id);
 
-	if (client != shooter->client) {
-		wl_resource_post_error(resource, WL_DISPLAY_ERROR_INVALID_OBJECT,
-				       "screenshooter failed: permission denied");
-		return;
-	}
+//	if (client != shooter->client) {
+//		wl_resource_post_error(resource, WL_DISPLAY_ERROR_INVALID_OBJECT,
+//				       "screenshooter failed: permission denied");
+//		return;
+//	}
 
 	wl_resource_set_implementation(resource, &screenshooter_implementation,
 				       data, NULL);
-- 
1.9.1

