diff --git a/screenshooter.c b/screenshooter.c
index d54b47d..b40171e 100644
--- a/screenshooter.c
+++ b/screenshooter.c
@@ -266,11 +266,11 @@ bind_shooter(struct wl_client *client,
 	resource = wl_resource_create(client,
 				      &screenshooter_interface, 1, id);
 
-	if (client != shooter->client) {
-		wl_resource_post_error(resource, WL_DISPLAY_ERROR_INVALID_OBJECT,
-				       "screenshooter failed: permission denied");
-		return;
-	}
+//	if (client != shooter->client) {
+//		wl_resource_post_error(resource, WL_DISPLAY_ERROR_INVALID_OBJECT,
+//				       "screenshooter failed: permission denied");
+//		return;
+//	}
 
 	wl_resource_set_implementation(resource, &screenshooter_implementation,
 				       data, NULL);
