--- a/rc
--- b/rc
@@ -26,12 +26,12 @@
         progress=$progress_size
     fi
     #echo "PROGRESS is $progress $runlevel $first_step + ($step of $num_steps)
-    #if type psplash-write >/dev/null 2>&1; then
-    #    TMPDIR=/mnt/.splash splash-write "PROGRESS $progress" || true
-    #fi
-    if [ -e /mnt/.psplash/psplash_fifo ]; then
-        echo "PROGRESS $progress" > /mnt/.psplash/psplash_fifo
+    if type psplash-write >/dev/null 2>&1; then
+        TMPDIR=/mnt/.splash splash-write "PROGRESS $progress" || true
     fi
+    #if [ -e /mnt/.psplash/psplash_fifo ]; then
+    #    echo "PROGRESS $progress" > /mnt/.psplash/psplash_fifo
+    #fi
 }


